### PR TITLE
refactor(evm): single-source the EIP-8037 intrinsic gas rules in the policy and harden their arithmetic

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
## Changes

EIP-8037's two-dimensional (regular + state) intrinsic-gas rules were computed in several places with two divergent definitions. This PR makes the gas policy the single owner of those rules and hardens their arithmetic. It supersedes and folds in three earlier draft PRs (#12243, #12245, #12246).

- **[fix] Guard the state-gas subtractions against `ulong` underflow.** `RevertRefundToHalt` reduced parent `StateGasUsed` with a raw subtraction relying on an unenforced invariant → now `SaturatingSub`, matching its sibling refund methods. `CreateAvailableFromIntrinsic` subtracted intrinsic regular + state gas under a `Debug.Assert` that is stripped in Release; a caller reaching it without validating intrinsic gas would wrap and receive an inflated budget → now a runtime guard returns an out-of-gas policy. (The reservoir-cap subtraction is left as-is: it is already covered by the unconditional `ExceedsCap` check in `ValidateStatic`.)
- **[test] Add a generic `GasPolicyContractTests<TGasPolicy>` conformance fixture.** `EthereumGasPolicy` is the only in-tree `IGasPolicy` implementation, but the interface is monomorphized to host others (downstream `ArbitrumGasPolicy`). The fixture pins the universal invariants a second implementation must uphold — it is the regression net the refactor commits rely on.
- **[refactor] Single-source the initial state-reservoir formula.** `txGasLimit − (intrinsicStateGas + DefaultTxGasLimitCap)` (saturating) was spelled out identically in `TransactionProcessor.CalculateInitialStateReservoir` and the `IGasPolicy` default `ComputeBlockRegularGas`; extracted into `Eip8037BlockGasInclusionCheck.CalculateInitialStateReservoir` and routed both callers through it.
- **[improvement] Compute the admission gas in one policy member.** The smallest `tx.GasLimit` that admits a transaction — `max(intrinsicRegular + intrinsicState, floor)` — was derived three ways: the state-aware `EthereumIntrinsicGas` converter (`TxValidator`, `GasEstimator`), an inline `spec.IsEip8037Enabled` branch in `ValidateStatic`, and — divergently — the state-*blind* `IntrinsicGas.MinimalGas`. Added `IntrinsicGas.StandardGas` and `IntrinsicGas.MinRequiredGasLimit` and routed all three admission sites through the latter. The state-blind `MinimalGas` is intentionally kept for the charging path, which reports only the regular component.
- **[refactor] Delete the `EthereumIntrinsicGas` converter.** Once the admission sites route through `MinRequiredGasLimit`, the converter — which re-flattened the policy's structured intrinsic gas back into a parallel `ulong` tuple — is production-dead. Removed it; the non-generic `IntrinsicGasCalculator.Calculate` now returns `IntrinsicGas<EthereumGasPolicy>`, and the remaining consumers (tests + `StateTestRunner`) read `StandardGas` / `FloorGas.Value` / `MinRequiredGasLimit` directly.

### Behaviour

Every path is byte-identical on validated transactions; `header.GasUsed = CombineBlockGas(Σ regular, Σ state)` is untouched. The admission routing collapses to `max(regular + state, floor)` at all three sites (EIP-8037: `max(R+S, max(R,floor)) = max(R+S, floor)` since `R+S ≥ R`; pre-8037 the state reservoir is 0, so both reduce to `max(R, floor)`). The two underflow guards only change unvalidated warmup/simulation paths, where a previously-undefined wrap becomes a defined clamped/out-of-gas result. The Optimism failed-deposit rule (`header.GasUsed += tx.GasLimit`) and the state-blind charging path are untouched.

## Types of changes

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Code refactoring (no functional changes, no API changes)
- [x] New feature (non-breaking; adds the conformance fixture and the admission-gas member)

## Testing

- [x] Unit + integration: `Nethermind.Evm.Test` (`EthereumGasPolicyTests`, `GasPolicyContractTests`, `Eip8037*`, `IntrinsicGasCalculatorTests`, `Eip2028/7623/7778/7928/7976/7981Tests`, `GasPriceExtractorTests`) and `Nethermind.Blockchain.Test` (`TransactionProcessorTests`, `Eip8037BlockGasIntegrationTests`, `TxValidatorTests`) green.
- [x] New tests: the two underflow regressions, the conformance fixture, the reservoir boundary cases, and the `MinRequiredGasLimit` state-aware-vs-state-blind divergence/equivalence tests.
- [x] EF `GeneralStateTests` (Amsterdam / EIP-8037 subset) via `Nethermind.Test.Runner`.
- [x] `dotnet format whitespace`.

## Notes

- Perf-neutral by construction: admission is per-tx (not per-opcode); the hot `EthereumGasPolicy` budget struct is unchanged; the added guard is one once-per-tx branch.
- Supersedes #12243 (→ underflow guard), #12245 (→ conformance fixture), #12246 (→ reservoir dedup), which are closed in favor of this consolidated PR.
- This targets `master`. The in-flight EIP repricings PR (#12214) targets `feature/eip-8246-v2`, not `master`, so there is no conflict here; the two reconcile at that branch's merge. That PR moves the state-gas fields to signed `long`, so these guards are re-expressed as `≥ 0` clamps at reconciliation time.

### CODE REVIEW

**Comment by benaadams** [comment]
Merge conflict

File: src/Nethermind/Nethermind.Blockchain/Tracing/GasEstimator.cs [hunk 1]
```diff
@@ -78,7 +79,7 @@ private EstimationResult EstimateInternal(
         if (CheckFunds(tx, spec, gasTracer, senderBalance, out UInt256 available) is { } fundsResult)
             return fundsResult;
 
-        ulong intrinsicGas = IntrinsicGasCalculator.Calculate(tx, spec, header.GasLimit).MinimalGas;
+        ulong intrinsicGas = EthereumGasPolicy.CalculateIntrinsicGas(tx, spec, header.GasLimit).MinRequiredGasLimit;
```
Comment (benaadams) [inline src/Nethermind/Nethermind.Blockchain/Tracing/GasEstimator.cs:82]: Should this be `EthereumGasPolicy` and not a `IntrinsicGas<EthereumGasPolicy>`?

Comment (AnkushinDaniil) [reply src/Nethermind/Nethermind.Blockchain/Tracing/GasEstimator.cs:82]: Intentional — `CalculateIntrinsicGas` returns `IntrinsicGas<EthereumGasPolicy>` to keep the Standard and Floor components separate; the EIP-8037 floor-gas and cap checks need both, and a collapsed `EthereumGasPolicy` would lose that split. Here we want the minimum gas *limit* the tx requires = `MinRequiredGasLimit`, which includes the state reservoir (`StandardGas = GetRemainingGas(Standard) + GetStateReservoir(Standard)`), unlike the state-blind `GetRemainingGas(MinimalGas)`. For non-8037 txs the reservoir is 0 so it equals the old minimal; for state-gas txs it's the correct floor — the `MinRequiredGasLimit_includes_state_reservoir_unlike_state_blind_minimal_gas` test pins exactly that distinction.

**Review by LukaszRozmej — APPROVED** [review]

**Review by wurdum — APPROVED** [review]

**Review by svlachakis — APPROVED** [review]
